### ISSUE TITLE
Coveralls shouldn't need access to private repos when used only for public repos

### ISSUE THREAD

**Issue by fritzo** [lemurheavy/coveralls-public#1585]
I am trying to configure coveralls for a public repo, but I cannot login without giving coveralls access to private repos as well. Can you please restrict reading to public repos?
![image](https://user-images.githubusercontent.com/648532/134434393-536b485d-9da0-4a18-9b4c-c2dd648542bc.png)

**Comment by afinetooth** [comment]
Hi @fritzo,

There are two (2) possible OAuth apps for Coveralls, `Coveralls` and `Coveralls Pro`. `Coveralls Pro` is the app that asks for private repo access, so you don't need to use that one. That app is usually engaged when you try to start a subscription for your user account/org account. To avoid that, simply don't enable any subscriptions and you'll still be able to access, and add, any of your public repos to coveralls from the [ADD REPOS](https://coveralls.io/repos/new) page.

If you have already authorized the `Coveralls Pro` app, you can simply go to Github and revoke access on it. Similarly, if you want to start from scratch, go to Github and revoke access for all apps with "Coveralls" in the name, then go to https://coveralls.io and login via Github OAuth. 

At that point, you will have OAuth based access to public scope only.

**Comment by fritzo** [comment]
Hi @afinetooth, the problem is I can't log into either of your links without giving coveralls access to private repos.  I don't recall having authorized `Coveralls Pro`; is there a way I can find out which repos it has access to and remove its access?  Could it be that, because I am a member of various github orgs, that if any of those orgs contain a single private repo that has granted access to `Coveralls Pro`, then I can no longer use public-only coveralls?  There are hundreds of repos among the github orgs in which I am a member.

**Comment by afinetooth** [comment]
Hi @fritzo, 

`Coveralls Pro` should only ever show you private repos for orgs that have a paid Coveralls subscription. If you're seeing any of your _own_ private repos, you either paid for a Coveralls subscription or are on a trial for a paid subscription.

You can always step back from the `Coveralls Pro` app and fall back to the `Coveralls` app, which only gives access to public repos. That can be done by canceling your paid subscription / trial, or by revoking access to the `Coveralls Pro` app at Github.

For full background, it works like this:

- When you first login to https://coveralls.io as a non-paying, general user, you authorize the `Coveralls` app, which gives you access to all of _your_ public repos, and the public repos for _orgs_ you belong to. 
- Then, when you subscribe to a paid Coveralls plan, or belong to a Github organization that has a paid subscription, you can hit the REFRESH PRIVATE REPOS button and authorize the `Coveralls Pro` app (if not already authorized through the subscription process), which gives you access to private repos for the given org(s) (only). The `Coveralls Pro` app then supersedes the `Coveralls` app as your main Coveralls integration. 
- Then, if you happen to cancel all subscriptions and fall back to general use---or in other words, revoke access to `Coveralls Pro`---Coveralls will leverage the `Coveralls` app again, maintaining access to public repos (only).
- Also worth knowing is that, if you ever delete / revoke the `Coveralls` app, you'll can simply login / OAuth again to https://coveralls.io, which authorizes the `Coveralls` app.

To address your specific questions:

> I don't recall having authorized `Coveralls Pro`; is there a way I can find out which repos it has access to and remove its access?

Unfortunately, you can't limit the app's access by repo. But you can limit the _orgs_ it has private access for. Just go to Github.com > Settings > Applications > Authorized OAuth Apps > `Coveralls Pro`, then Grant or Revoke access by org/user. (To revoke, you'll need to revoke the entire app and re-authorize just for the orgs you want private access to.)

(In fact, granting access to other orgs doesn't really grant access to their private repos because Coveralls won't attempt access to those repos unless the org has a Coveralls subscription.)

> Could it be that, because I am a member of various github orgs, that if any of those orgs contain a single private repo that has granted access to Coveralls Pro, then I can no longer use public-only coveralls? 

No. First, you need to authorize the `Coveralls Pro` app yourself. Belonging to an org with a paid Coveralls subscription won't automatically authorize the `Coveralls Pro` app for your Github account. 

That said, in order to gain access to any of the _org's_ private repos, you _will_ have to authorize the Coveralls Pro app, because only that app manages connections to private repos.

If, however, you _do_ authorize `Coveralls Pro`, the access it grants will only be to _that org's_ private repos. 

(Furthermore, through Coveralls, you will only have access to the org's private repos that _you_, _personally_, have access to at Github. If you don't have access to one of the org's repos through Github, you won't have access to it through Coveralls.)

In summary, to address what I imagine is your main concern, authorizing `Coveralls Pro` will not suddenly grant access to _your_ private repos if you don't, personally, have a paid Coveralls account.

Hope that all makes sense.

What may be confusing you is that, if you _do_ have the `Coveralls Pro` app installed, Coveralls will leverage _that_ as your main integration with Github, just because it provides _both_ public and private repo access. (But only private access to repos of orgs with paid accounts.)

If you remove `Coveralls Pro` you'll only ever have access to public repos, even if some private repos are available to you through orgs you belong to.

**Comment by joebowbeer** [comment]
@afinetooth Do you have plans to support GitHub App-based authorization for organizations with private repos?

**Comment by afinetooth** [comment]
Hi, @joebowbeer. Replied to your [related question](https://github.com/lemurheavy/coveralls-public/issues/1622#issuecomment-1100456937), [here](https://github.com/lemurheavy/coveralls-public/issues/1622#issuecomment-1101552218).

**Comment by rsteube** [comment]
Got the same issue. `coveralls pro` is constantly requesting access to private repos. Removed the `coveralls` auth and after authorizing `coveralls` again i instantly get a request from `coveralls pro` and thus don't have access to the account.

**Comment by afinetooth** [comment]
Hi, @rsteube. Yes, **Coveralls Pro** is our OAuth app that has private scope (access to private repos). The **Coveralls** Oauth app has public scope only.

What are you trying to do exactly (or trying to avoid). Hopefully I can help.

**Comment by rsteube** [comment]
@afinetooth just trying to sign in (https://coveralls.io/sign-in) and i am getting the request from `coveralls pro` instead of `coveralls`. I might be doing sth. wrong though.

**Comment by afinetooth** [comment]
@rsteube, sorry for the delay.

I looked at your account and it appears you currently only have access to public repos at Github. Which is to say, it appears you have only approved access for the **Coveralls** Github OAuth app (as opposed to our **Coveralls Pro** Github OAuth App).

I understand it's possible that the orgs you have access to only have public repos, but I don't see that any of those orgs have subscriptions to coveralls.io, which means there would be no private scope access to private repos at github.com.

If you did, at one time, approve the **Coveralls Pro** Github OAuth App, since it's no longer associated with one of your orgs, it would not provide private repo access to any org's repos.

But for good measure, you can always remove / revoke access for the **Coveralls Pro** app with the following steps:

1. Login to github.com
2. Go to Your Account > **Settings**
3. Go to[**Applications**](https://github.com/settings/applications)
4. Go to **Authorized OAuth Apps**
5. Find Coveralls Pro
6. Click on [...] menu
7. Choose **Revoke**

Closing for now since this runs across a couple of use cases. If anyone has an issue that sounds similar, please still open a [New issue](https://github.com/lemurheavy/coveralls-public/issues/new).
